### PR TITLE
backend handler for checking in and submitting Requests

### PR DESCRIPTION
**Pull request by Raulp8**
structure for sending post requests to add a rating to a store

### CODE REVIEW

File: shopsafe-backend/src/main/java/com/google/sps/servlets/CheckInServlet.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+import com.google.appengine.api.datastore.KeyFactory.Builder;
+
+
+/** Servlet that handles user's rating of a store*/
+//TODO: determine url
+@WebServlet("/checkin")
+public class CheckInServlet extends HttpServlet {
+
+
+    @Override
```
Comment (caroljli) [inline shopsafe-backend/src/main/java/com/google/sps/servlets/CheckInServlet.java]: Should probably delete the empty GET function

Comment (glstewart17) [inline shopsafe-backend/src/main/java/com/google/sps/servlets/CheckInServlet.java]: You can remove the get request for now.

File: shopsafe-backend/src/main/java/com/google/sps/data/StoreDatastoreHandler.java [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+ */
+public class StoreDatastoreHandler {
+    
+    private String id;
+    private Key key;
+    private DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+    public StoreDatastoreHandler(String id) {
+        this.id = id;
+        this.key = new Builder("Store", id).getKey();
```
Comment (glstewart17) [inline shopsafe-backend/src/main/java/com/google/sps/data/StoreDatastoreHandler.java:43]: Does this get the key, and create one, if this is a new id?

File: shopsafe-backend/src/main/java/com/google/sps/data/StoreDatastoreHandler.java [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+            ArrayList<String> previousRatingIds = (ArrayList<String>) storeEntity
+                .getProperty("ratings");
+            String ratingEntityIdName = ratingEntity.getKey().getName();
+            previousRatingIds.add(ratingEntityIdName);
+            storeEntity.setProperty("ratings", previousRatingIds);
+
+            datastore.put(ratingEntity);
+            datastore.put(storeEntity);
+    }
+    /*
```
Comment (glstewart17) [inline shopsafe-backend/src/main/java/com/google/sps/data/StoreDatastoreHandler.java:173]: Add another line.

File: shopsafe-backend/src/main/java/com/google/sps/data/StoreDatastoreHandler.java [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+        @SuppressWarnings("unchecked")
+            ArrayList<String> previousRatingIds = (ArrayList<String>) storeEntity
+                .getProperty("ratings");
+            String ratingEntityIdName = ratingEntity.getKey().getName();
+            previousRatingIds.add(ratingEntityIdName);
+            storeEntity.setProperty("ratings", previousRatingIds);
+
+            datastore.put(ratingEntity);
+            datastore.put(storeEntity);
+    }
```
Comment (glstewart17) [inline shopsafe-backend/src/main/java/com/google/sps/data/StoreDatastoreHandler.java:171]: newStore and updateStore need more comments. It took me a bit to understand the process by which you are storing the maps.

**Review by glstewart17 — APPROVED** [review]
Other than improving the readability, it looks good to me. I'm approving, but you might want to resolve these before merging.

File: shopsafe-backend/src/main/java/com/google/sps/servlets/CheckInServlet.java (hunk 1 above)
Comment (glstewart17) [reply shopsafe-backend/src/main/java/com/google/sps/servlets/CheckInServlet.java]: Too late, I already commented that. XD

File: shopsafe-backend/src/main/java/com/google/sps/data/StoreDatastoreHandler.java [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+            ratingEntity.setProperty(ratingField, Double.parseDouble(ratingValue[0]));
+        });
+        return ratingEntity;
+    }
+
+    /*
+     * Places new store along with rating
+     */
+    private void newStore(Map<String, String[]> ratingsMap) {
+
```
Comment (caroljli) [inline shopsafe-backend/src/main/java/com/google/sps/data/StoreDatastoreHandler.java]: Could get rid of the extra spaces for styling

**Review by caroljli — APPROVED** [review]

File: shopsafe-backend/src/main/java/com/google/sps/data/StoreDatastoreHandler.java (hunk 5 above)
Comment (caroljli) [reply shopsafe-backend/src/main/java/com/google/sps/data/StoreDatastoreHandler.java]: I agree with Gabe, you should probably fix the styling and add a little more commenting before merging :)

File: shopsafe-backend/src/main/java/com/google/sps/data/StoreDatastoreHandler.java (hunk 2 above)
Comment (Raulp8) [reply shopsafe-backend/src/main/java/com/google/sps/data/StoreDatastoreHandler.java:43]: yeah it creates a new key each time, but that doesnt mean it creates a new datastore entry, its just a way to communicate with datastore

**Pull request merged**